hack/gke/xfs-formatter/xfs-checker.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -euExo pipefail
+shopt -s inherit_errexit
+
+raid_device="$( grep -E "/mnt/stateful_partition/kube-ephemeral-ssd($| )" /proc/mounts | sed -E -e 's/([^ ]+) .+/\1/' )"
+
+if [[ -z "${raid_device}" ]]; then
+  if [[ -f "${RAID_DEVICE_FILE}" ]]; then
+    raid_device="$( head -n 1 "${RAID_DEVICE_FILE}" )"
+  fi
+  if [[ -z "${raid_device}" ]]; then
+    echo "Raid device not mounted"
+    exit 255
+  fi
+fi
+
+echo "${raid_device}" > "${RAID_DEVICE_FILE}"
```

Comment:
should this be written only if we have autodetected the device and not when we read it from the same file?